### PR TITLE
Add tool choice options to MessageParameter

### PR DESCRIPTION
**Pull request by ahammouda**
I'm pretty new to swift, so patience is much appreciated 🙏

### CODE REVIEW

File: Sources/Anthropic/Service/AnthropicService.swift [hunk 1]
```diff
@@ -124,7 +124,7 @@ extension AnthropicService {
       guard let httpResponse = response as? HTTPURLResponse else {
          throw APIError.requestFailed(description: "invalid response unable to get a valid HTTPURLResponse")
       }
-      printHTTPURLResponse(httpResponse)
+      printHTTPURLResponse(httpResponse, data: data)
```
Comment (ahammouda) [inline Sources/Anthropic/Service/AnthropicService.swift:127]: This should probably be wrapped in a `#if DEBUG ... `

Comment (ahammouda) [reply Sources/Anthropic/Service/AnthropicService.swift:127]: nvm - I forgot the method already does this

File: Sources/Anthropic/Public/Parameters/Message/MessageParameter.swift [hunk 2]
```diff
@@ -77,7 +77,9 @@ public struct MessageParameter: Encodable {
    ///
    /// **cacheControl**: Prompt Caching
    let tools: [Tool]?
-   
+    
+   let toolChoice: ToolChoice?
```
Comment (ahammouda) [inline Sources/Anthropic/Public/Parameters/Message/MessageParameter.swift:81]: Looks like you'll want to add a docstring and update the README as well

File: Sources/Anthropic/Public/Parameters/Message/MessageParameter.swift [hunk 3]
```diff
@@ -211,6 +213,22 @@ public struct MessageParameter: Encodable {
...
+
+   public struct ToolChoice: Codable {
+      public enum ToolType: String, Codable {
+         case tool
+         case auto
+         case any
+      }
+      
+      let type: ToolType
+      let name: String
```
Comment (fumito-ito) [inline Sources/Anthropic/Public/Parameters/Message/MessageParameter.swift:225]: The `name` parameter is only used when `ToolType` is set to `tool`. Shouldn’t we prevent it from being specified when ToolType is set to `auto` or `any` ?

https://docs.anthropic.com/en/docs/build-with-claude/tool-use#forcing-tool-use

**Review by jamesrochabrun — APPROVED** [review]
Thanks! We need to set the name to be optional but I can do that as follow up PR.

File: Sources/Anthropic/Public/Parameters/Message/MessageParameter.swift (hunk 3 above)
Comment (jamesrochabrun) [reply Sources/Anthropic/Public/Parameters/Message/MessageParameter.swift:225]: Thanks for flagging, will take care of this in a follow up PR

**Pull request merged**